src/test/ui/type/type-check/assignment-expected-bool.stderr

```diff
@@ -63,6 +63,10 @@ error[E0308]: mismatched types
 LL |     let _: bool = if { 0 = 0 } {
    |                        ^^^^^ expected `bool`, found `()`
    |
+help: you might have meant to use pattern matching
+   |
+LL |     let _: bool = if { let 0 = 0 } {
```

Comment:
these help messages don't seem worth it :thinking: not sure how to best suppress them though

Reply:
These were a side effect of the change that didn't seem worth it to spend too much time removing.

Reply:
Can we check that the LHS is an identifier? I think ideally we would also not emit the diagnostics when surrounded in `{}`. Would it be that much trouble to fix?

Reply:
We can, but the lhs on an assignment can be _any_ pattern in theory, which is why I didn't bother on adding that check. I'll try to implement that soon.

Reply:
I now gate against literals in particular, because we _do_ want to suggest when someone writes `if Option::Some(0) = x`, for example.